fedbiomed/researcher/aggregators/scaffold.py

```diff
@@ -244,18 +244,26 @@ def update_correction_states(
         # c_i^{t+1} = delta_i^t + (x^t - y_i^t) / (M * eta)
         for node_id, updates in model_updates.items():
             d_i = self.nodes_deltas[node_id]
-            self.nodes_states[node_id] = {
-                key: d_i[key] + val / (self.nodes_lr[node_id][idx] * n_updates)
-                for idx, (key, val) in enumerate(updates.items())
-            }
+            for (key, val) in updates.items():
+
+                if self.nodes_lr[node_id].get(key):
```

Comment:
```suggestion
                if self.nodes_lr[node_id].get(key) is not None:
```

Reply:
Yes you are right about this one!  if `self.nodes_lr[node_id].get(key)` returns 0, it should pass that statement